### DISCUSSION TITLE
Looking for feedback for sensible defaults for crepe pitch detection

### DISCUSSION THREAD

**Discussion by BWagener** [rakuri255/UltraSinger#88] [Ideas]
[torchcrepe](https://github.com/maxrmorrison/torchcrepe), the library used for pitch detection, supports detecting frequencies between 0-2006Hz. torchcrepe runs with a range from 50-2006Hz by default (UltraSinger currently uses these values) and allows changing the lower and upper limits.

What should be the defaults for these limits? The idea is the better suited the range is to the domain the more accurate the results should be.

**Comment by DerGitRob** [comment]
the entire range of men’s and women’s voices remains between about 65 Hz for a male with a very deep bass voice to the highest note of a female coloratura soprano, just above 1,000 Hz, at 1,280 Hz. A female high-pitched scream can go quite a bit higher, to around 3,000 Hz.
Source: https://www.axiomaudio.com/blog/audio-oddities-frequency-ranges-of-male-female-and-childrens-voices

the frequency range of human hearing, which is responsible for the perception of speech. It covers the frequencies from 300 to 3000 Hz
Source: https://seaindia.in/blogs/human-voice-frequency-range/

**Reply by BWagener** [reply]
That's great!

One part I didn't understand is the frequency range for perception of speech. The way I get it is that this would mean any frequency below 300Hz is not perceived by the ear.

On another note, I have a test file which is pitched incorrectly when running torchcrepe with a frequency range of 50-2006Hz, but is pitched correctly with a range of 0-2006Hz. https://github.com/rakuri255/UltraSinger/pull/87#discussion_r1319843636

So for now I think it would be best to consider the full range torchcrepe supports until someone can properly test it.
